src/diagnostic-queue.ts

```diff
@@ -106,6 +106,7 @@ export class DiagnosticEventQueue {
         const uri = this.client.toResource(file).toString();
         const diagnosticsForFile = this.diagnostics.get(uri);
         diagnosticsForFile?.onDidClose();
+        this.diagnostics.delete(uri);
```

Comment:
This PR does fix the issue that I have experienced. Thanks.

---

One question I have is if you want to match the VS Code behavior?
 
I compared the behavior of VS Code and this PR.

VS code will clear the diagnostic as soon as file_a is closed.

In this PR, 
the diagnostics will not be cleared until the file_a is open again.

Which makes me think if the diagnostics should be published to the client on did close, 
if you want to match with VS Code behavior?

Reply:
yeah, the intention of `onDidClose` was to clear diagnostics before closing. should be fixed now?

Reply:
I would not move the call of this.publishDiagnostics() to a different place. The previous position was ok.

But I would swap 109 line with 108,
So that `this.diagnostics.delete(uri);` is executed first before onDidClose is called.

Wdyt?

Reply:
I'm not sure how would that help. Wouldn't `FileDiagnostics` instance still publish diagnostics that were reported at the time of closing the file?

(I'm looking at this while trying to do real work so I might be a bit distracted)

Reply:
> (I'm looking at this while trying to do real work so I might be a bit distracted)

Same, and I am even writing the review on my phone. I will leave this out for later today... :⁠-⁠)

Reply:
I was wrong.